### ISSUE TITLE
Clicking on the Switch component does nothing in Solid JS

### ISSUE THREAD

**Issue by jackvane** [cschroeter/park-ui#358]
**Current Version**: "@ark-ui/solid": "^3.4.1"

When assigning controlled state to the switch, clicking on the switch does not change state. If you remove all props from the Switch component, clicking on the switch still does nothing.

```jsx
import { action, useSubmission } from "@solidjs/router";
import { Show, createEffect, createSignal } from "solid-js";
import { Switch } from "~/components/ui/switch";

const login = action(async (formData: FormData) => {
  "use server";

  ...

}, "login");

export default function Page() {
  const [showPlainText, setShowPlainText] = createSignal<boolean>(false);

  return (
    <>
      <div>
        <form action={login} class="space-y-8" method="post">

...

          <div>
            <Switch
              size="sm"
              checked={showPlainText()}
              onCheckedChange={e => {
                console.log(e.checked); // does nothing
                setShowPlainText(e.checked); // does nothing
              }}
            >
              Show One Time Password
            </Switch>
          </div>

          <Button type="submit">Sign in with One Time Password</Button>
        </form>
      </div>
    </>
  );
}

```

I am using the default switch as per the docs:

```jsx
import { Switch as ArkSwitch, type SwitchRootProps } from "@ark-ui/solid";
import { Show, children, splitProps } from "solid-js";
import { type VariantProps, tv } from "tailwind-variants";

export interface SwitchProps extends SwitchRootProps, SwitchVariantProps {}

export const Switch = (props: SwitchProps) => {
  const [variantProps, switchProps] = splitProps(props, ["size", "class"]);
  const [localProps, rootProps] = splitProps(switchProps, ["children"]);
  const getChildren = children(() => localProps.children);
  const { root, control, label, thumb } = styles(variantProps);

  return (
    <ArkSwitch.Root class={root()} {...rootProps}>
      <ArkSwitch.Control class={control()}>
        <ArkSwitch.Thumb class={thumb()} />
      </ArkSwitch.Control>
      <Show when={getChildren()}>
        <ArkSwitch.Label class={label()}>{getChildren()}</ArkSwitch.Label>
      </Show>
    </ArkSwitch.Root>
  );
};

type SwitchVariantProps = VariantProps<typeof styles>;

const styles = tv(
  {
    base: "switchRecipe",
    defaultVariants: { size: "md" },
    slots: {
      root: "switchRecipe__root",
      label: "switchRecipe__label",
      control: "switchRecipe__control",
      thumb: "switchRecipe__thumb"
    },
    variants: {
      size: {
        sm: {
          root: "switchRecipe__root--size_sm",
          label: "switchRecipe__label--size_sm",
          control: "switchRecipe__control--size_sm",
          thumb: "switchRecipe__thumb--size_sm"
        },
        md: {
          root: "switchRecipe__root--size_md",
          label: "switchRecipe__label--size_md",
          control: "switchRecipe__control--size_md",
          thumb: "switchRecipe__thumb--size_md"
        },
        lg: {
          root: "switchRecipe__root--size_lg",
          label: "switchRecipe__label--size_lg",
          control: "switchRecipe__control--size_lg",
          thumb: "switchRecipe__thumb--size_lg"
        }
      }
    }
  },
  { twMerge: false }
);

```

**Comment by dannylin108** [comment]
I use `onChange`, not `onCheckedChange` and it works 

```typescript

const HSwitch = () => {
    const { t } = useLocale()!;
    const { settings, setSettings } = useAppSettings()!;

    return <HStack justifyContent='space-between'>
        <Text fontWeight='medium'>{t.settings.h24()}</Text>
        <Switch checked={settings.time24h}
            onChange={() => setSettings('time24h', !settings.time24h)}
        />
    </HStack>;
}
```

**Comment by Lenstack** [comment]
Add  `<ArkSwitch.HiddenInput/>` to 
![imagen](https://github.com/cschroeter/park-ui/assets/42081538/86e25c49-50c8-4397-9450-da85e326da28)

**Comment by spd789562** [comment]
Yes, a lot of form base component rely on `Ark`'s input, and it not on Park UI docuument, if some component not even working, just check `Ark UI` document, it would help a lot cuz `Park UI` rely on it
like https://ark-ui.com/solid/docs/components/switch

**Comment by cschroeter** [comment]
@jackvane 

The issue has been resolved.

https://github.com/cschroeter/park-ui/assets/1846056/94e6938a-21f9-45ab-b291-370c1c65168e

**Comment by lilouartz** [comment]
> The issue has been resolved.

Still having this issue in v3.5.0

**Comment by lilouartz** [comment]
This is my code:

```
import { type Assign } from '@ark-ui/react';
import {
  Switch as ArkSwitch,
  type SwitchRootProps,
} from '@ark-ui/react/switch';
import { css, cx } from '#app/styled-system/css';
import { splitCssProps } from '#app/styled-system/jsx';
import {
  switchRecipe,
  type SwitchRecipeVariantProps,
} from '#app/styled-system/recipes';
import { type JsxStyleProps } from '#app/styled-system/types';
import { forwardRef } from 'react';

export type SwitchProps = {} & Assign<JsxStyleProps, SwitchRootProps> &
  SwitchRecipeVariantProps;

export const Switch = forwardRef<HTMLLabelElement, SwitchProps>(
  (props, ref) => {
    const [variantProps, switchProps] = switchRecipe.splitVariantProps(props);
    const [cssProps, localProps] = splitCssProps(switchProps);
    const { children, className, ...rootProps } = localProps;
    const styles = switchRecipe(variantProps);

    return (
      <ArkSwitch.Root
        className={cx(styles.root, css(cssProps), className)}
        ref={ref}
        {...rootProps}
      >
        <ArkSwitch.Control className={styles.control}>
          <ArkSwitch.Thumb className={styles.thumb} />
        </ArkSwitch.Control>
        {children && (
          <ArkSwitch.Label className={styles.label}>{children}</ArkSwitch.Label>
        )}
        <ArkSwitch.HiddenInput />
      </ArkSwitch.Root>
    );
  },
);

Switch.displayName = 'Switch';

```

**Comment by lilouartz** [comment]
After some troubleshooting, I think this is an issue with react-compiler. Disabling react-compiler made the switch input work again.

**Comment by cschroeter** [comment]
I've released new and improved snippets. Please double check with https://park-ui.com/react/docs/components/switch

**Comment by lilouartz** [comment]
@cschroeter There is a minor issue in the new generated files:

```
/app/components/ui/primitives/text.tsx
  46:24  error  'as' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

**Comment by lilouartz** [comment]
@cschroeter Had to do a bit of post-processing to make it work,

```
"generate:park-ui": "park-ui components add --all && replace-in-file '/styled-system//g' '#app/styled-system/' 'app/components/ui/**/*.tsx' --isRegex && replace-in-file '~/lib/create-style-context' '#app/components/ui/create-style-context' 'app/components/ui/**/*.tsx' && eslint --fix app/components/ui",
```

Maybe the above is helpful in hinting at ways that the config for park CLI could be improved.

**Comment by cschroeter** [comment]
I've resolved an issue where `libDir` was not correctly resolved based on the user's configuration in `park-ui.json`
The fix is available in `v0.7.1`
